src/Diagnostics.DataProviders/DataProviders/Base/SupportObserverDataProviderBase.cs

```diff
@@ -101,8 +105,14 @@ protected async Task<string> GetObserverResource(string url, string resourceId =
                 request.Headers.TryAddWithoutValidation("Authorization", await GetToken(resourceId));
             }
 
-            var cancelToken = new CancellationToken();
+            var cancelTokenSource = new CancellationTokenSource();
+            var cancelToken = cancelTokenSource.Token;
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
             var response = await _httpClient.SendAsync(request, cancelToken);
+            stopwatch.Stop();
+            Logger.LogDataProviderMessage(RequestId, "ObserverDataProvider", $"url:{url};resourceId:{resourceId};latency:{stopwatch.Elapsed};statusCode:{response.StatusCode}");
```

Comment:
- Remove the resourceId from the logging as that is only AAD related.
- The latency is tracked by `DataProviderLogDecorator`